### PR TITLE
Fix incorrect deprecations

### PR DESCRIPTION
**Pull request by namusyaka**
ref #1344

### CODE REVIEW

File: lib/grape/router/route.rb [hunk 1]
```diff
@@ -56,6 +57,15 @@ def route_path
         pattern.path
       end
 
+      def hidden
```
Comment (dblock) [inline lib/grape/router/route.rb]: Are we really just trying to make grape-swagger happy here? The whole `hidden` thing is not a Grape feature, just happens to be an option that grape-swagger "extends". Maybe it's better to expose `options` instead and let grape-swagger deal with it?

**Comment by namusyaka** [comment]
@dblock Thanks for your opinion. I've just tried to reflect it to my changes.

**Comment by dblock** [comment]
I am happy with this as is, squashed and CHANGELOGed.

**Comment by dblock** [comment]
I also think the code needs comments of what `settings` vs. `options` is.

**Comment by namusyaka** [comment]
@dblock Could you review this updates? I also squashed :)

**Comment by dblock** [comment]
Looks good to me, feel free to merge on green if I don't get to it.

**Pull request merged**